tools/crm_resource_runtime.c

```diff
@@ -965,39 +965,24 @@ generate_resource_params(pe_resource_t *rsc, pe_node_t *node,
 
 bool resource_is_running_on(pe_resource_t *rsc, const char *host)
 {
-    bool found = TRUE;
-    GList *hIter = NULL;
-    GList *hosts = NULL;
+    bool retval = false;
+    GList *lst = NULL;
 
-    if(rsc == NULL) {
-        return FALSE;
-    }
-
-    rsc->fns->location(rsc, &hosts, TRUE);
```

Comment:
This call might make them different.

All resource types use native_location() as their location method (it probably shouldn't be a method, but methods are public API, so ...). native_location() recurses for any resource children, but the other function doesn't.

I'm not sure whether that has an effect that has in practice, i.e. whether a group/clone/bundle's running_on is different from the union of its children's running_on. Of course it's also possible we should recurse in the other function anyway.

Reply:
I did not expect that we'd have any relevant tests, but changing pe__rsc_running_on_any_node_in_list to use the recursive method results in:
```
--- /home/clumens/src/pacemaker/cts/scheduler/summary/stop-failure-no-quorum.summary    2021-06-10 16:43:01.350262232 -0400
+++ /home/clumens/src/pacemaker/cts/scheduler/summary/stop-failure-no-quorum.summary.pe 2021-06-16 15:53:22.342131647 -0400
@@ -41,5 +41,8 @@
   * Full List of Resources:
     * Clone Set: dlm-clone [dlm]:
       * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
+    * Clone Set: clvm-clone [clvm]:
+      * clvm   (lsb:clvmd):     FAILED pcmk-3 (UNCLEAN, blocked)
+      * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
     * ClusterIP        (ocf:heartbeat:IPaddr2):         Stopped
     * Fencing  (stonith:fence_xvm):     Stopped
--- /home/clumens/src/pacemaker/cts/scheduler/summary/stop-failure-no-fencing.summary   2021-06-10 16:43:01.414262784 -0400
+++ /home/clumens/src/pacemaker/cts/scheduler/summary/stop-failure-no-fencing.summary.pe        2021-06-16 15:53:22.414132273 -0400
@@ -9,6 +9,9 @@
   * Full List of Resources:
     * Clone Set: dlm-clone [dlm]:
       * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
+    * Clone Set: clvm-clone [clvm]:
+      * clvm   (lsb:clvmd):     FAILED pcmk-3 (UNCLEAN, blocked)
+      * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
     * ClusterIP        (ocf:heartbeat:IPaddr2):         Stopped
 
 Transition Summary:
@@ -24,4 +27,7 @@
   * Full List of Resources:
     * Clone Set: dlm-clone [dlm]:
       * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
+    * Clone Set: clvm-clone [clvm]:
+      * clvm   (lsb:clvmd):     FAILED pcmk-3 (UNCLEAN, blocked)
+      * Stopped: [ pcmk-1 pcmk-2 pcmk-3 pcmk-4 ]
     * ClusterIP        (ocf:heartbeat:IPaddr2):         Stopped
```

Reply:
In that CIB, clvm had a stop failure on pcmk-3, so the scheduler has to assume it could be active, which means it's reasonable that crm_mon considers it active as well. However it's odd that it shows in the Stopped list as well; I'd expect it to show as active or stopped, but not both.

Reply:
Though, one possibility is this is now the correct behavior and not having those resources displayed before was a bug.

Reply:
The code in lib/pengine/clone.c that generates stopped_list also only checks rsc->running_on, so it seems likely to me that there will be confusion if it's not using the same test as pe__rsc_running_on_any_node_in_list (man, that's a long name to type) uses.

Reply:
> Though, one possibility is this is now the correct behavior and not having those resources displayed before was a bug.

That's what I was suggesting, it's reasonable to treat (and show) clvm on pcmk-3 as an active resource (i.e. as shown here), not an inactive resource (i.e. hidden as it was before).

It does make sense that the Stopped code should use the inverse of the active test.

Reply:
Heh, okay, kind of playing "Is it a bug?" whack-a-mole here.  I've modified the test for producing the stopped list to use the same criteria which makes some things better in cts-scheduler output and some things only possibly better.  I'm going to turn them into patches and add to this PR.